node.gyp

```diff
@@ -289,7 +289,7 @@
           '-Wl,-bnoerrmsg',
         ],
       }],
-      ['(OS=="linux" or OS=="mac") and llvm_version!=0', {
+      ['llvm_version==0 or (llvm_version!=0 and OS=="linux")', {
```

Comment:
I'm having trouble understanding what problem this fixes. This is adding `-latomic` for all platforms when gcc is used, why? It is also removing `-latomic` on Macs when clang is used, but wasn't one of the bugs in the linked reports explicitly about needing `-latomic` with clang on Mac?

Reply:
technically clang always needs libatomic, but node only uses it on mac and linux for the v8 sigsegv handlers.

I don't believe it's ever needed on GCC, but I could be wrong.

Reply:
>This is adding -latomic for **all platforms when gcc** is used, why?

I assumed this was implicitly  required by the previous state. I've replace it with: 
```
OS=="linux" and llvm_version!=0
```
to only add it on cllang+linux
>but wasn't one of the bugs in the linked reports explicitly about needing -latomic with clang on Mac?

no, the https://github.com/nodejs/node/issues/28231 mention clang+linux, it fails on clang+darwin https://github.com/NixOS/nixpkgs/pull/64253#issuecomment-508385537
>technically clang always needs libatomic

not quite, `libatomic` is a gcc library. Clang by default is to be built to use it on Linux, and to use `compiler-rt` on darwin